src/main/java/com/google/graphgeckos/dashboard/storage/DataRepository.java

```diff
@@ -0,0 +1,66 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.graphgeckos.dashboard.storage;
+
+/**
+ * An interface for providing unified I/O to the storage component.
+ */
+public interface DataRepository {
+  /**
+   * Creates a new database entry of the "revision" type with the metadata of the revision.
+   * If there already is an entry with the same commit hash, ignores the request.
+   *
+   * @param entryData a ParsedGitData instance, must have a non-null "commitHash" field
+   * @throws IllegalArgumentException if entryData or it's "commitHash" field is null
+   */
+  void createRevisionEntry(ParsedGitData entryData) throws IllegalArgumentException;
+
+  /**
+   * Updates an existing revision's database entry, with the individual information from
+   * a particular buildbot. If there is no entry associated with the provided commit hash,
+   * ignores the request.
+   *
+   * @param updateData a ParsedBuildbotData instance, must have a non-null "commitHash" field
+   * @throws IllegalArgumentException if updateData or it's "commitHash", "builderName",
+   *     or "logs" fields are null.
```

Comment:
I'm not sure if this is something that you should change right now, but is looks like the invariants of these data objects are well distributed over the code base. Is this because they are protos and therefore can't have their own code to check invariants? Sometimes in those cases it makes sense to wrap the protos in an object that maintains the invariants. I can't tell if that makes sense here, though.

Reply:
So, if this is a manually defined class, then it might make sense to have a more abstract concept of the state the object needs to be in than just defining a few fields. Maybe you want a concept like `isStorable()` or something similar on the object, and then here you just refer to that concept. My concern was mostly about the long list of individual properties where it's not inherently clear why you need exactly those properties. What happens if you add new properties to the object? Do they need to be added here? It's often nicer if you have a higher-level abstraction that is implemented and validated in a central place.

Again, not sure if this makes sense here. Use your best judgement.

Reply:
Regarding the `isStorable()` idea, both #43 and #44, where each individual implementation of the classes are, have the `validUpdateData()` and `validCreateData()`. Perhaps it makes more sense to refer to the failure of these method calls, instead of each individual field?

Reply:
Yes, that makes sense.

Reply:
I just changed that. Thank you!